packages/shared/src/contexts/OnboardingContext.tsx

```diff
@@ -138,10 +148,26 @@ export const OnboardingContextProvider = ({
       onCloseOnboardingModal,
       onShouldUpdateFilters: setShouldUpdateFilters,
       onInitializeOnboarding,
+      shouldSkipIntro,
     }),
-    [isOnboarding, user, alerts, onboardingMode, shouldUpdateFilters],
+    [
+      isOnboarding,
+      user,
+      alerts,
+      onboardingMode,
+      shouldUpdateFilters,
+      shouldSkipIntro,
+    ],
   );
 
+  // if onboarding is done/closed reset
+  // state values to default
+  useEffect(() => {
+    if (!isOnboarding) {
+      setSkipIntro(false);
+    }
+  }, [isOnboarding]);
```

Comment:
We should avoid the usage of `useEffect` whenever possible. Let's probably create a function to wrap the calling of `setIsOnboarding(false)` to also execute `setSkipIntro(false)` at the same time. This would also benefit us once we move to React 18.

Reply:
For state syncing like this, moving it to event handler is better I agree.